memtx: record getting the index size in the MVCC (#10652)
tarantool/tarantool: changelogs/unreleased/gh-10149-index-len-mvcc.md

## bugfix/memtx

> * Fixed a bug when getting the index size wasn't recorded in the
>   MVCC and so its change didn't lead to the transaction conflict

Reviewer: ```suggestion
* Fixed a bug when the initial index size wasn't recorded in the
  MVCC, so no transaction conflict occurred if the index size changed
  during the transaction
```
